fog/view/server/src/db_fetcher.rs

```diff
@@ -280,6 +297,14 @@ impl<DB: RecoveryDb + Clone + Send + Sync + 'static> DbFetcherThread<DB> {
 
         for (ingress_key, block_index) in next_block_index_per_ingress_key.into_iter() {
             // Attempt to load data for the next block.
+            if !self.sharding_strategy.should_process_block(block_index) {
+                log::trace!(
+                    self.logger,
+                    "Not processing block_index {} because this shard is not responsible for it.",
+                    block_index,
+                );
```

Comment:
I think that, just skipping blocks that aren't in our shard is right, but we need to also tell the block tracker something about this, otherwise, the block tracker will think we are still blocked on this block.

I think its okay to just call `block_tracker.block_pressed` at this line also

Reply:
I made a new PR #2387 that makes this change (and returns the minimum highest observed block count to the client).